### PR TITLE
Combine all agents into one ci.yml

### PR DESCRIPTION
**Pull request by merwanehamadi**
### PR Quality Checklist
- [X] I have run the following commands against my code to ensure it passes our linters:
    ```shell
    black .
    isort .
    mypy .
    autoflake --remove-all-unused-imports --recursive --ignore-init-module-imports --ignore-pass-after-docstring --in-place agbenchmark
    ```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
-name: Python CI
+name: CI
 
 on:
+  workflow_dispatch:
+    branches: [ master ]
+  schedule:
+    - cron: '0 8 * * *'
```
Comment (merwanehamadi) [inline .github/workflows/ci.yml:7]: We can trigger the CI to test the agents against OpenAI or wait every 8 AM UTC

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -107,3 +121,61 @@ jobs:
...
+            docker-compose up -d --build
+          else
+            echo "Unknown agent name: $AGENT_NAME"
+            exit 1
+          fi
+          
+          pip install ../../dist/*.whl
+
+          if [ "${GITHUB_EVENT_NAME}" == "schedule" ] || [ "${GITHUB_EVENT_NAME}" == "workflow_dispatch" ]; then
+            agbenchmark start --maintain
```
Comment (merwanehamadi) [inline .github/workflows/ci.yml:157]: this makes the call to the LLM

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -107,3 +121,61 @@ jobs:
...
+            exit 0
+            agbenchmark start --maintain --mock
+            agbenchmark start --improve --mock
+            agbenchmark start --mock
+            agbenchmark start --mock --category=retrieval
+            agbenchmark start --mock --category=regression
+            agbenchmark start --mock --category=interface
+            agbenchmark start --mock --category=code
+            agbenchmark start --mock --category=memory
+            agbenchmark start --mock --category=memory --category=code
```
Comment (merwanehamadi) [inline .github/workflows/ci.yml:168]: I am testing our features by just using them

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -107,3 +121,61 @@ jobs:
...
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+          AGENT_NAME: ${{ matrix.agent-name }}
+          PROMPT_USER: false # For mini-agi. TODO: Remove this once mini-agi follows the standards.
+
+      - name: Upload logs as artifact
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: gpt-engineer-projects
+          path: agent/gpt-engineer/projects
```
Comment (merwanehamadi) [inline .github/workflows/ci.yml:181]: right now I only added the logs for gpt-engineer.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -107,3 +121,61 @@ jobs:
...
+            source venv/bin/activate
+            pip install -r requirements.txt
+          elif [ "$AGENT_NAME" == "SuperAGI" ]; then
+            cp config_template.yaml config.yaml
+            sed -i 's/OPENAI_API_KEY:.*/OPENAI_API_KEY: "'"${{ secrets.OPENAI_API_KEY }}"'"/' config.yaml
+            docker-compose up -d --build
+          else
+            echo "Unknown agent name: $AGENT_NAME"
+            exit 1
+          fi
```
Comment (merwanehamadi) [inline .github/workflows/ci.yml:152]: I will put these into a file called: 
setup_auto-gpt.sh

and then dynamically call the right setup depending on the agent.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -68,10 +73,18 @@ jobs:
...
   tests:
-
+    name: ${{ matrix.agent-name }}
     runs-on: ubuntu-latest
     env:
       min-python-version: "3.10"
+    strategy:
+      fail-fast: false
+      matrix:
+        agent-name:
```
Comment (SilenNaihin) [inline .github/workflows/ci.yml:83]: you forgot SuperAGI here

**Review by SilenNaihin — CHANGES_REQUESTED** [review]

**Review by SilenNaihin — APPROVED** [review]

**Pull request merged**